src/createWidgetBase.ts

```diff
@@ -151,19 +151,19 @@ const createWidget: WidgetBaseFactory = createStateful
 				return v(tag, this.getNodeAttributes(), this.getChildrenNodes());
 			},
 
-			set children(this: Widget<WidgetProperties>, children: DNode[]) {
+			get children(this: Widget<WidgetProperties>) {
+				return widgetInternalStateMap.get(this).children;
+			},
+
+			setChildren(this: Widget<WidgetProperties>, children: DNode | DNode[]): void {
 				const internalState = widgetInternalStateMap.get(this);
-				internalState.children = children;
+				internalState.children = Array.isArray(children) ? children : [ children ];
```

Comment:
While this is of course fine, two thoughts come to mind when I see this pattern:

1. Should we have a core arrayUtil method to do things like this (probably not as it's probably more expensive, but it might be cleaner, especially if something like this is eventually added to the language.
2. Are we assuming things are arrays that should also support iterables (probably doesn't matter here in this example, but in general throughout Dojo 2, are we adequately supporting iterables or assuming arrays).